tests/test_plotting.py

```diff
@@ -60,581 +60,489 @@ def test_kwarg_warning(DummyStellarator):
 def test_kwarg_future_warning(DummyStellarator):
     """Test that passing in deprecated kwargs throws a warning."""
     eq = load(load_from=str(DummyStellarator["output_path"]))
-    with pytest.warns(FutureWarning):
-        fig, ax = plot_surfaces(eq, zeta=2)
+    with pytest.raises(FutureWarning):
+        with warnings.catch_warnings():
+            warnings.simplefilter("error")
+            fig, ax = plot_surfaces(eq, zeta=2)
     return None
 
 
-@pytest.mark.unit
-@pytest.mark.mpl_image_compare(remove_text=True, tolerance=tol_1d)
-def test_1d_p():
-    """Test plotting 1d pressure profile."""
-    eq = get("SOLOVEV")
-    fig, ax, data = plot_1d(eq, "p", figsize=(4, 4), return_data=True)
-    assert "p" in data.keys()
-    return fig
+class TestPlot1D:
+    """Tests for plot_1d."""
 
+    @pytest.mark.unit
+    @pytest.mark.mpl_image_compare(remove_text=True, tolerance=tol_1d)
+    def test_1d_p(self):
+        """Test plotting 1d pressure profile."""
+        eq = get("SOLOVEV")
+        fig, ax, data = plot_1d(eq, "p", figsize=(4, 4), return_data=True)
+        assert "p" in data.keys()
+        return fig
 
-@pytest.mark.unit
-@pytest.mark.mpl_image_compare(remove_text=True, tolerance=tol_1d)
-def test_1d_elongation():
-    """Test plotting 1d elongation as a function of toroidal angle."""
-    eq = get("precise_QA")
-    grid = LinearGrid(N=20, NFP=eq.NFP)
-    fig, ax, data = plot_1d(
-        eq, "a_major/a_minor", grid=grid, figsize=(4, 4), return_data=True
-    )
-    assert "a_major/a_minor" in data.keys()
-    return fig
-
-
-@pytest.mark.unit
-def test_1d_fsa_consistency():
```

Comment:
Redundant with `plot_1d_iota_radial` for fixed current equilibrium

Reply:
This test was added in #376 for issue #368. Since the git diff is ugly, for convenience here is the test on master: https://github.com/PlasmaControl/DESC/blob/5801ea8238a45b6f6674cc0d2ebe67d9d598242f/tests/test_plotting.py#L92

 `test_1d_iota_radial` tests whether the plot of the rotational transform of DSHAPE_current changes. It fails to test whether the plot is accurate in the first place. Prior to #376, `iota` would be computed (inaccurately) on a 1d grid, which that pull request corrects so that it is computed on a 2d grid, yet the baseline image for `iota` does not change for `test_1d_iota_radial`.

The correction is measurable with quantities computed in `test_1d_fsa_consistency`, i.e. magnetic well as reported in the issue.

If the purpose of `test_1d_iota_radial` is solely to test whether an 2d grid is used for flux surface average quantities, we should remove `test_1d_iota_radial` and keep `test_1d_fsa_consistency`. If the rotational transform image is an important test in particular, we should keep both tests, or merge them.

Reply:
I think they're both somewhat redundant, since the logic in `eq.compute` now ensures that the correct grid is used for computing FSA quantities (and there are a number of other tests that check that, eg `test_compute_everything`). I kept `iota_radial` just to exercise that bit of the code, and its faster than `1d_fsa_consistency`

Reply:
Ok but we no longer have a test that confirms whether the plotting is done on the correct grid. Because test_1d_iota_radial will pass if the computation is done on a 1d grid too, and test_compute_everything doesn't check how we compute what we plot.

Reply:
I think it should be fine, since `plotting._compute` just calls `eq.compute` which includes logic for always using the correct grid.